File: src/main/java/iesvdm/adrian/proyecto/domain/Usuario.java
package iesvdm.adrian.proyecto.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;
import iesvdm.adrian.proyecto.domain.Partido;

import java.util.ArrayList;
import java.util.Collection;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@EqualsAndHashCode(of = "id")
@Builder
@Table(name = "usuario")
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_usuario")
    private Long id;

    @NotBlank
    private String nombre;

    @NotBlank
    private String apellido1;

    @NotBlank
    private String apellido2;

    @NotBlank
    private String usuario;

    @NotBlank
    private String password;

    private String rol;

    @Email
    private String email;

    @Size(min = 9, max = 9)
    private String telefono;
    @JsonIgnore
    @ManyToOne
    private Equipo equipo;

}

File: crud-front/src/app/model/usuario.ts
import { Equipo } from "./equipo";

export class Usuario {
    id!: number;
    nombre: string;
    apellido1: string;
    apellido2: string;
    usuario: string;
    password: string;
    rol: string;
    email: string;
    telefono: string;
    equipo: Equipo;

    constructor(nombre: string, apellido1: string, apellido2: string, usuario: string, password: string, rol: string, email: string, telefono: string, equipo: Equipo){
        this.nombre = nombre;
        this.apellido1 = apellido1;
        this.apellido2 = apellido2;
        this.usuario = usuario;
        this.password = password;
        this.rol = rol;
        this.email = email;
        this.telefono = telefono;
        this.equipo = equipo;
    }

}
